src/server/api/routers/project.ts

```diff
@@ -1,33 +1,40 @@
 import { z } from "zod";
 import { createTRPCRouter, protectedProcedure } from "@/server/api/trpc";
 import { accessCheck } from "@/server/api/routers/access";
-import { PutObjectCommand } from "@aws-sdk/client-s3";
-import { getSignedUrl } from "@aws-sdk/s3-request-presigner";
-import crypto from "crypto";
 import { env } from "@/env";
+import { getUploadURL } from "@/server/s3";
 
 const allowedFileTypes = ["image/png", "image/jpeg"];
 const maxFileSize = 1048576; // 1MB
+const projectValidation = z.object({
+  title: z.string().min(1).max(64),
+  username: z
+    .string()
+    .min(1)
+    .max(64)
+    .regex(/^[a-zA-Z0-9-_]+$/),
```

Comment:
```suggestion
    .regex(/^[a-z0-9-]+$/),
```
match the slugify function - uppercase letters not necessary, and underscores aren't best practice afaik